### PR TITLE
[AAH-3103] Add useful links

### PR DESCRIPTION
**Pull request by nixocio**
Refactor Namespace details page

Modify Namespace form

Add PageDetailKeyValue component

Details Page

image

Create

image

### CODE REVIEW

File: frontend/hub/namespaces/PageInputDynamicFields.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { Button, Flex, FlexItem, TextInput } from '@patternfly/react-core';
+import { PlusCircleIcon, TrashIcon } from '@patternfly/react-icons';
+import { useFieldArray, useFormContext, Controller } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { PageFormSection } from '../../../framework/PageForm/Utils/PageFormSection';
+
+export function PageInputDynamicFields() {
```
Comment (nixocio) [inline frontend/hub/namespaces/PageInputDynamicFields.tsx:7]: This name will be updated. Just to have an idea what direction this work was going.

**Review by ShaiahWren — COMMENTED** [review]
Nice work @nixocio!
A couple surprises I encountered. Do you see these?
1. Surprising red when pasting a LINK. Why is that? It has no message and no network error. Must be a ui issue.
Screenshot 2024-03-04 at 10 53 35 AM

2. On full screen list view, the link is clipped off. The formatting is also surprising. We can fix by putting it in a new section.I will look for a solution today. :)

Screenshot 2024-03-04 at 10 56 01 AM

File: framework/PageDetails/PageDetailKeyValueList.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      <DescriptionListTerm data-cy={'key-value-list-title'}>{title}</DescriptionListTerm>
+      <DescriptionList isHorizontal style={{ paddingLeft: '20px' }}>
+        <DescriptionListGroup>
+          <DescriptionListTerm>{keyColumn}</DescriptionListTerm>
+          <DescriptionListDescription>{valueColumn}</DescriptionListDescription>
+        </DescriptionListGroup>
+        {keyValue.map((item, index) => (
+          <DescriptionListGroup key={index}>
+            <DescriptionListDescription
+              style={{ whiteSpace: 'nowrap', textOverflow: 'ellipsis' }}
```
Comment (jamestalton) [inline framework/PageDetails/PageDetailKeyValueList.tsx:39]: Would we want this to use ellipsis vs wrapping?

File: framework/PageDetails/PageDetailKeyValueList.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+              data-cy={`item-key-${index}`}
+            >
+              {item.key}
+            </DescriptionListDescription>
+            <DescriptionListDescription
+              style={{ whiteSpace: 'nowrap', overflow: 'hidden', textOverflow: 'ellipsis' }}
```
Comment (jamestalton) [inline framework/PageDetails/PageDetailKeyValueList.tsx:45]: Would we want this to use ellipsis vs wrapping?

File: framework/PageDetails/PageDetailKeyValueList.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  const { keyValue, title, keyColumn, valueColumn, renderValue } = props;
+
+  const defaultRenderValue = (item: KeyValue) => <div data-cy={`item-value`}>{item.value}</div>;
+
+  const renderFunction = renderValue || defaultRenderValue;
+
+  return (
+    <>
+      <DescriptionListTerm data-cy={'key-value-list-title'}>{title}</DescriptionListTerm>
+      <DescriptionList isHorizontal style={{ paddingLeft: '20px' }}>
```
Comment (jamestalton) [inline framework/PageDetails/PageDetailKeyValueList.tsx:31]: padding left here is not optimal at small screens. might be better to have a section with heading and no left padding.

**Review by jamestalton — APPROVED** [review]

**Comment by nixocio** [comment]
> Nice work @nixocio! A couple surprises I encountered. Do you see these?
> 
>     1. Surprising red when pasting a LINK. Why is that? It has no message and no network error. Must be a ui issue.
> 
> 
> Screenshot 2024-03-04 at 10 53 35 AM
> 
>     2. On full screen list view, the link is clipped off. The formatting is also surprising. We can fix by putting it in a new section.I will look for a solution today. :)
> 
> 
> Screenshot 2024-03-04 at 10 56 01 AM

@ShaiahWren thanks for your review, great findings, please take another look.

**Comment by ShaiahWren** [comment]
Nice work @nixocio , I'm still seeing some ux inconsistencies with the other items on the detail page. I think the "Useful links" probably shouldn't be an h3 because that indicates a tertiary view for the screen reader and I'm not sure that's appropriate here. If the intent is to get a smaller text, we should use font-size=14px to match the other headers (name, desc, company, etc). 
Screenshot 2024-03-06 at 9 49 51 AM

**Pull request merged**